aesm-client/src/win_aesm_client.rs

```diff
@@ -0,0 +1,359 @@
+    >,
+    report_attestation_status: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            platform_info: *mut u8,
+            platform_info_size: u32,
+            attestation_status: u32,
+            update_info: *mut u8,
+            update_info_size: u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_ps_cap: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            ps_cap: *mut u64,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    sgx_register: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            white_list_cert: *mut u8,
+            white_list_cert_size: u32,
+            registration_data_type: u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    proxy_setting_assist: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            proxy_info: *mut u8,
+            proxy_size: u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    query_sgx_status: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            sgx_status: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_whitelist_size: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            white_list_size: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_white_list: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            white_list: *mut u8,
+            buf_size: u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_sec_domain_id: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            sec_domain_id: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    switch_sec_domain: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            sec_domain_id: u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_epid_provision_status: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            epid_pr_status: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_platform_service_status: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            pse_status: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+}
+
+const CLSID_AESMINTERFACE: CLSID = CLSID {
+    Data1: 0x82367CAB,
+    Data2: 0xF2B9,
+    Data3: 0x461A,
+    Data4: [0xB6, 0xC6, 0x88, 0x9D, 0x13, 0xEF, 0xC6, 0xCA],
+};
+const IID_IAESMINTERFACE: IID = IID {
+    Data1: 0x50AFD900,
+    Data2: 0xF309,
+    Data3: 0x4557,
+    Data4: [0x8F, 0xCB, 0x10, 0xCF, 0xAB, 0x80, 0x2C, 0xDD],
+};
+
```

Comment:
Add function:

```rust
use winapi::shared::winerror::SUCCEEDED;

trait HresultExt {
    fn into_io_error(self) -> Result<Self, IoError>;
}

impl HresultExt for HRESULT {
    fn into_io_error(self) -> Result<HRESULT, IoError> {
        if SUCCEEDED(self) {
            Ok(self)
        } else {
            Err(IoError::from_raw_os_error(self))
        }
    }
}
```

Use this everywhere you have a COM function that returns an HRESULT.